csharp-algorithms/QuickSort/QuickSortTests/QuickSortUnitTests.cs

```diff
@@ -0,0 +1,46 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using QuickSort;
+using System;
+using System.Diagnostics;
+
+namespace QuickSortTests
+{
+    [TestClass]
+    public class QuickSortUnitTests
+    {
+        [TestMethod]
+        public void GivenUnsortedArray_ThenReturnSortedArray()
+        {
+            var array = new int[] { 73, 57, 49, 99, 133, 20, 1 };
+            var expected = new int[] { 1, 20, 49, 57, 73, 99, 133 };
+            var sortFunction = new QuickSortMethods();
+
+            var sortedArray = sortFunction.SortArray(array, 0, array.Length - 1, string.Empty);
+
+            Assert.IsNotNull(sortedArray);
+            CollectionAssert.AreEqual(sortedArray, expected);
+        }
+
+        [TestMethod]
+        public void GivenUnsortedArray_CheckInstanceType()
+        {
+            var sortFunction = new QuickSortMethods();
+            var array = QuickSortMethods.CreateRandomArray(200);
+
+            var sortedArray = sortFunction.SortArray(array, 0, array.Length - 1, string.Empty);
+
+            Assert.IsInstanceOfType(sortedArray, typeof(int[]));
+        }
+
+        [TestMethod]
+        public void GivenUnsortedArray_CheckSortedNotNull()
+        {
+            var sortFunction = new QuickSortMethods();
+            var array = QuickSortMethods.CreateRandomArray(200);
+
+            var sortedArray = sortFunction.SortArray(array, 0, array.Length - 1, string.Empty);
+
+            Assert.IsNotNull(sortedArray);
+        }
```

Comment:
GivenUnsortedArray_WhenArrayIsSorted_ThenResultingArrayIsNotNull
For example ^^
Remember, the "when" part is the action and it's mandatory, that's what you test.
And you sometimes forget the "then" in the "then" word. (the second and the third test)